lib/internal/Magento/Framework/Css/PreProcessor/Adapter/CssInliner.php

```diff
@@ -54,6 +54,16 @@ public function disableStyleBlocksParsing()
         $this->emogrifier->disableStyleBlocksParsing();
     }
 
+    /**
+     * @param bool $debug
+     *
+     * @return void
+     */
+    public function setDebugMode($debug)
+    {
+        $this->emogrifier->setDebug($debug);
```

Comment:
Let's not expose too much methods when we don't really need it.

Please discard changes in `app/code/Magento/Email/Model/Template/Filter.php` and instead of introducing a new method here inject `AppState` and do `setDebug(true)` in constructor in case of developer mode.

Reply:
On it.

Reply:
Done and repushed. I'm not very familiar with how Magento handles DI, so I'm keeping my fingers crossed.